CSharpSourceCode/CampaignSupport/WanderersController.cs

```diff
@@ -0,0 +1,85 @@
+﻿using HarmonyLib;
+using System.Collections.Generic;
+using System.Linq;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.Actions;
+using TaleWorlds.CampaignSystem.SandBox.CampaignBehaviors;
+using TaleWorlds.Core;
+using TOW_Core.Utilities.Extensions;
+
+namespace TOW_Core.CampaignSupport
+{
+    public class WanderersController : CampaignBehaviorBase
+    {
+        private List<Hero> _companions;
+        private Dictionary<Settlement, CampaignTime> _companionSettlements;
+
+        public override void SyncData(IDataStore dataStore)
+        {
+        }
+
+        public override void RegisterEvents()
+        {
+            CampaignEvents.OnAfterSessionLaunchedEvent.AddNonSerializedListener(this, OnAfterSessionLaunched);
+            CampaignEvents.AfterSettlementEntered.AddNonSerializedListener(this, OnAfterSettlementEntered);
+        }
+
+        private void OnAfterSessionLaunched(CampaignGameStarter gameStarter)
+        {
+            var urbanBehavior = Campaign.Current.GetCampaignBehavior<UrbanCharactersCampaignBehavior>();
+            if (urbanBehavior != null)
+            {
+                _companions = Traverse.Create(urbanBehavior).Field("_companions").GetValue<List<Hero>>();
```

Comment:
accessing private fields like this should be avoided if there is another way. Furthermore, creating a private _companions field in this class and moving wanderers to it will make it out of sync with the one in Urbanbehaviour. I don't think we need it at all. More comments further down.